src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/ResourceMonitoringExtensions.cs

```diff
@@ -88,24 +84,114 @@ public static class ResourceMonitoringExtensions
         _ = Throw.IfNull(builder);
         _ = Throw.IfNull(section);
 
-        return builder.ConfigureTrackerInternal(configure: optionsBuilder => optionsBuilder.Bind(section));
+        return builder.ConfigureMonitorInternal(optionsBuilder => optionsBuilder.Bind(section));
+    }
+
+    /// <summary>
+    /// Registers a publisher that emits resource utilization information as a Windows performance counter.
+    /// </summary>
+    /// <param name="builder">The builder instance used to configure the tracker.</param>
+    /// <returns>The value of <paramref name="builder"/>.</returns>
+    /// <exception cref="ArgumentNullException"><paramref name="builder"/> is <see langword="null" />.</exception>
+    public static IResourceMonitorBuilder AddWindowsPerfCounterPublisher(this IResourceMonitorBuilder builder)
+    {
+        _ = Throw.IfNull(builder);
+
+        var os = Environment.OSVersion;
+        if (os.Platform != PlatformID.Win32NT)
+        {
+            throw new NotSupportedException("AddWindowsPerfCounterPublisher is only available on Windows.");
+        }
+
+        return builder.AddPublisher<WindowsPerfCounterPublisher>();
     }
 
-    private static IServiceCollection AddResourceUtilizationInternal(
+    private static IServiceCollection AddResourceMonitoringInternal(
         this IServiceCollection services,
         Action<IResourceMonitorBuilder> configure)
     {
-        configure.Invoke(new ResourceUtilizationBuilder(services));
+        var builder = new ResourceMonitorBuilder(services);
+
+#if NETFRAMEWORK
+        _ = builder.AddWindowsProvider();
+#elif !NET8_0_OR_GREATER
+        var os = Environment.OSVersion;
+        if (os.Platform == PlatformID.Win32NT)
+        {
+            _ = builder.AddWindowsProvider();
+        }
+        else if (os.Platform == PlatformID.Unix)
+        {
+            _ = builder.AddLinuxProvider();
+        }
+        else
+        {
+            throw new NotSupportedException("Resource monitoring is not supported on this operating system.");
+        }
```

Comment:
Is there a reason we can't use this for .NET8+ too?

Reply:
I'm not sure what you mean. Can you restate please?